addons/mail/static/src/models/message/message.js

```diff
@@ -497,6 +497,20 @@ function factory(dependencies) {
             return false;
         }
 
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeMessageTypePrefix() {
+            if (this.message_type === 'notification') {
+                return this.env._t("System notification by ");
+            }
+            if (!this.is_discussion && !this.is_notification) {
+                return this.env._t("Note by ");
+            }
+            return this.env._t("Message by ");
```

Comment:
For proper translation the name should be injected into the translation with `%s`. Current way does not allow to change words order if necessary.

It was already a problem in the template, but now that it is moved in JS, we can actually resolve it.
That would require being able to tell the reader to not read the name twice though.

Reply:
Now this text is not a prefix but a message type and it's located at the message itself. That way, we don't need to deal with neither words order nor hiding author name a 2nd time (which would be complicated because, as mentioned above, `aria-hidden` has also CSS meaning).

Field was renamed to `messageType` to reflect its new purpose.

Reply:
`messageType` as name is too close to `message_type`, I think it is confusing.
Also tests don't pass now.

But it's still a problem to be translated. It is not a message type text if it is followed by "by". Clearly this indicates that the name of the author should follow, and in languages where the name should be somewhere else grammatically on the phrase than at the end, it will not be translatable correctly.

Reply:
Ok I didn't see the diff, you actually removed the "by", fine then. Just need a better name for the field.

Reply:
Suggestion for name: `messageTypeText`? Which would already bit a bit better.

Reply:
Done, renamed to messageTypeText